File: Servidor_java/Modelos/Pelicula.java
package Servidor_java.Modelos;

public class Pelicula extends Produto{

    private String modelo;
    private String material;
    

    public Pelicula(){
        super();
    }

    public Pelicula(int id,String nome,String descricao,double preco,int estoque,String modelo,String material){

        super(id,nome,descricao,preco,estoque);

        this.modelo = modelo;
        this.material = material;
    }

    public String getModelo(){return modelo;}
    public String getMaterial(){return material;}

    public void setModelo(String modelo){this.modelo = modelo;}
    public void setMaterial(String material){this.material = material;}

    public String toString() {
        return "Pelicula [id=" + getId() + ", nome=" + getNome() + ", preco=" + getPreco() + ", estoque=" + getEstoque() + "modelo="+ modelo + "material="+ material +"]";
    }


}

File: Cliente_cpp/src/modelos/Pelicula.cpp
#include "modelos/Pelicula.h"
#include <string>
#include <iostream>

Pelicula::Pelicula(int id, std::string nome, std::string descricao, float preco, int quantidade_estoque, std::string modelo, std::string material)
    : Produto(id, nome, descricao, preco, quantidade_estoque), modelo(modelo), material(material) {}

std::string Pelicula::getModelo() const {
    return modelo;
}

void Pelicula::setModelo(const std::string& modelo) {
    this->modelo = modelo;
}

std::string Pelicula::getMaterial() const {
    return material;
}

void Pelicula::setMaterial(const std::string& material) {
    this->material = material;
}

void Pelicula::mostrar() const {
    Produto::mostrar();
    std::cout << "Modelo: " << modelo << std::endl;
    std::cout << "Material: " << material << std::endl;
}